Remove Windows arm32 support

BruceForstall opened the pull request:
Mostly, remove the CI runs. Windows arm32 product support was removed before .NET 6.

BruceForstall committed:
Remove Windows arm32 support

Mostly, remove the CI runs. Windows arm32 product support was removed
before .NET 6.

BruceForstall commented:
/azp run runtime

BruceForstall commented:
cc @dotnet/jit-contrib

BruceForstall commented:
Comments, anyone?

There are probably many more things that could be removed after this is merged. The primary goal here was to stop running win-arm builds and tests in the CI system.

jkoritzinsky commented on eng/pipelines/coreclr/templates/build-job.yml:
```diff
@@ -202,10 +202,6 @@ jobs:
       - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) $(archType) -hostarch x64 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
         displayName: Build CoreCLR Cross-Arch Tools (Tools that run on x64 targeting x86)
 
-    - ${{ if and(eq(parameters.osGroup, 'windows'), eq(parameters.archType, 'arm')) }}:
-      - script: $(Build.SourcesDirectory)/src/coreclr/build-runtime$(scriptExt) $(buildConfig) $(archType) -hostarch x86 $(osArg) -ci $(compilerArg) -component crosscomponents -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" $(officialBuildIdArg) $(clrRuntimePortableBuildArg)
-        displayName: Build CoreCLR Cross-Arch Tools (Tools that run on x86 targeting arm)
-
```
There's a condition in `eng/Subsets.props` that corresponds to this condition. Do we want to remove that as well?

BruceForstall replied:
Yes. But that will be a follow-up.

jkoritzinsky reviewed (approved):
LGTM. Do we want to remove support for building the product build for Windows ARM32, or just the runtime test build/run?

jkotas commented:
There is more that can be cleaned up in the sources (e.g. delete src\coreclr\vm\arm\*.asm). It can be done as a separate change.

trylek commented on src/tests/run.cmd:
```diff
@@ -200,7 +199,7 @@ echo.
 echo.
 echo./? -? /h -h /help -help   - View this message.
-echo ^<build_architecture^>      - Specifies build architecture: x64, x86, arm, or arm64 ^(default: x64^).
+echo ^<build_architecture^>      - Specifies build architecture: x64, x86, or arm64 ^(default: x64^).
```
Hmm, I'm still seeing linux_arm / linux_musl_arm jobs in the above pipelines, are we removing arm32 support even on these?

BruceForstall replied:
No, linux-arm is still a supported platform. (But run.cmd is only used on Windows)

trylek commented on src/tests/issues.targets:
```diff
@@ -291,235 +291,6 @@
-        </ExcludeList>
-    </ItemGroup>
-
     <!-- Windows arm64 specific excludes -->
```
Not as part of this PR but I guess at some point we should audit this issue list and close those that are no longer relevant i.e. that only occur on the no longer supported Windows arm platform.

trylek reviewed (approved):
LGTM overall modulo one nitpick. I guess we'll likely want to make an equivalent change in the .NET 6 / 7 servicing branches, correct?

BruceForstall commented:
> I guess we'll likely want to make an equivalent change in the .NET 6 / 7 servicing branches, correct?

Since we don't support win-arm in .NET 6/7, that would make sense.

BruceForstall committed:
Restore a few files

Only change YML files to remove windows_arm runs.

Don't build Windows arm cross components.

Don't touch mono files.

Changing scripts to remove builds, and remove code, can follow.

BruceForstall commented:
I decided to simplify this to only remove win-arm CI runs. Removing the ability to build via build scripting can happen later.

BruceForstall commented:
/azp run runtime
